tripy/tests/frontend/test_shape.py

```diff
@@ -40,13 +40,27 @@ def other_values(request):
 
 
 class TestShapeScalar:
-    @pytest.mark.parametrize("value", [1, tp.Tensor(1), np.array(2)])
+    @pytest.mark.parametrize(
+        "value",
+        [
+            1,
+            tp.Tensor(1),
+            # Note: if we don't specify the dtype, the tensor constructor will insert a cast
+            # and the assert below about the trace_tensor's producer will fail.
+            np.array(2, dtype=np.int32),
+        ],
+    )
     def test_scalar_shape(self, value):
-        s = tp.ShapeScalar(values)
+        s = tp.ShapeScalar(value)
 
         assert isinstance(s, tp.ShapeScalar)
         assert s.trace_tensor.producer.inputs == []
 
+        if isinstance(value, int):
+            assert s.__str__() == f"shape_scalar({value})"
+        else:
+            assert s.__str__() == f"shape_scalar({int(value.tolist())})"
```

Comment:
Can we add a separate test for this? I don't think it should matter how the `ShapeScalar` was constructed, so we don't need to parameterize it over that.

Reply:
Done